amulet/level/formats/anvil_forge_world.py

```diff
@@ -33,5 +40,24 @@ def game_version_string(self) -> str:
         except Exception:
             return f"Java Forge Unknown Version"
 
+    def _register_modded_dimensions(self):
+        for region_path in glob.glob(
+            os.path.join(glob.escape(self.path), "dimensions", "*", "**", "region"),
+            recursive=True,
+        ):
+            dim_path = os.path.dirname(region_path)
+            child_dir_names = set(
+                basename
+                for basename in os.listdir(dim_path)
+                if os.path.isdir(os.path.join(dim_path, basename))
+            )
+            if not {"region"}.issubset(child_dir_names):
```

Comment:
Since you only check one directory it should be.
The os.path.dirname in the last line was also wrong.
```py
    def _register_modded_dimensions(self):
        for region_path in glob.glob(
            os.path.join(glob.escape(self.path), "dimensions", "*", "**", "region"),
            recursive=True,
        ):
            if not os.path.isdir(region_path):
               continue
            dim_path = os.path.dirname(region_path)
            rel_dim_path = os.path.relpath(dim_path, self.path)
            _, namespace, *base_name = rel_dim_path.split(os.sep)

            dimension_name = f"{namespace}:{'/'.join(base_name)}"
            self._register_dimension(rel_dim_path, dimension_name)
```

Reply:
I'll add this in tomorrow when I get the chance!